sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/TimeWindow.scala

```diff
@@ -107,7 +108,7 @@ object TimeWindow {
       throw new IllegalArgumentException(
         s"Intervals greater than a month is not supported ($interval).")
     }
-    cal.microseconds
+    cal.days * CalendarInterval.MICROS_PER_DAY + cal.microseconds
```

Comment:
@cloud-fan here we still use a "1 day = 24 hours" assumption. is this OK?

Reply:
here we use interval as a duration, and it's safe to use `1 day = 24 hours` (this is always true).